Source/WebGPU/WebGPU/SwapChainCoreAnimation.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021-2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#import "SwapChain.h"
+
+#import <QuartzCore/QuartzCore.h>
+
+namespace WebGPU {
+
+class Device;
+
+class SwapChainCoreAnimation : public SwapChain {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    static Ref<SwapChainCoreAnimation> create(const Device& device, const WGPUSwapChainDescriptor& descriptor, CAMetalLayer *layer)
+    {
+        return adoptRef(*new SwapChainCoreAnimation(device, descriptor, layer));
+    }
+
+    virtual ~SwapChainCoreAnimation();
+
+    TextureView* getCurrentTextureView() override;
+    void present() override;
+
+private:
+    SwapChainCoreAnimation(const Device&, const WGPUSwapChainDescriptor&, CAMetalLayer *);
+
+    CAMetalLayer *m_layer { nil };
```

Comment:
Similar to the comment https://github.com/WebKit/WebKit/pull/8665/files#r1070535343 I think CAMetalLayer should be in SurfaceCoreAnimation and not here

Reply:
This was done earlier.